lib/process.ml

```diff
@@ -1,88 +1,169 @@
-type exit_reason = Normal | Exit_signal | Bad_link | Exception of exn
-type signal = Message
+type exit_reason =
+  | Normal
+  | Exit_signal
+  | Bad_link
+  | Link_down of Pid.t
+  | Exception of exn
+
+type signal = Message | Exit of exit_reason | Link of Pid.t | Monitor of Pid.t
 
 module Messages = struct
   type monitor = Process_down of Pid.t
   type Message.t += Monitor of monitor | Exit of Pid.t * exit_reason
 end
 
 type state = Runnable | Waiting | Running | Exited of exit_reason
-type process_flags = { mutable trap_exits : bool }
+type process_flags = { trap_exits : bool Atomic.t }
 type process_flag = Trap_exit of bool
 
-let default_flags = { trap_exits = false }
+let default_flags () = { trap_exits = Atomic.make false }
```

Comment:
this bit me so hard 😭 – whenever we'd turn on the logger, processes would start trapping exits.